src/main/java/Node.java

```diff
@@ -0,0 +1,98 @@
+import java.util.*;
+
+public class Node implements Comparable<Node>
+{
+    private SubwayStations.Station station;
+    private double distance = Integer.MAX_VALUE; //distance from source node
+    private List<SubwayStations.Station> shortestPath = new ArrayList<>();
+    private List<Node> neighbors = new ArrayList<>();
+    private Node previous;
```

Comment:
I don't know why you need both `previous` and `shortestPath`

Reply:
It was a mistake to have both. I originally was using `shortestPath` but then switched to `previous` and just forgot to remove `shortestPath` afterwards.